src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adler(num) {
+    if (num === undefined) {
+      const add = sum;
+
+      sum = 0;
+
+      return add;
+    };
+
+    sum += num;
```

Comment:
Avoid using a for loop to calculate the sum of an array, use 'reduce' instead. Replace this line to sum the arguments using 'reduce'.